### PR TITLE
Add a test for lambdify used with numba 

### LINKED ISSUE #20070: Add a test that lambdify works with numba
We should have a test in our test suite that tests that lambdify() works properly if you jit it with `numba.jit`. If we have that, then numba can add it to their integration tests to avoid potentially breaking it in the future (and we can also avoid breaking it too). 

See https://github.com/numba/numba/issues/2328

### PR DESCRIPTION
**Pull request by sumit-158**
#### References to other Issues or PRs
Fixes #20070
#### Brief description of what is fixed or changed
Adding test in [test_lumbidify.py](https://github.com/sympy/sympy/blob/master/sympy/utilities/tests/test_timeutils.py) that work for numba 

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by anutosh491** [comment]
> #### Other comments
> #### Release Notes
> NO ENTRY

You need to add release notes correctly for the checks to pass , if you're not sure how to do so refer to [guide on how to write release notes](https://github.com/sympy/sympy/wiki/Writing-Release-Notes)

**Comment by anutosh491** [comment]
Also line `1356` has some trailing whitespaces which need to be removed .python bin/strip_whitespace followed by the file name 
`python bin/strip_whitespace sympy/utilities/tests/test_lambdify.py` is the command which would hep you here !

**Comment by sumit-158** [comment]
> Thank you for the Help!!

**Comment by anutosh491** [comment]
You would also need to update your name in the `./mailmap` file , otherwise the tests would fail . I'll attach a string of comments by one of the maintainers here , it seems to be a good guide for this

**Comment by sumit-158** [comment]
@oscarbenjamin @oscargus Please review this PR.

**Comment by oscargus** [comment]
This is the outcome:

```
test_issue_20070 numba not installed s
```
so numba should be installed as well. That is done by adding numba here:
https://github.com/sympy/sympy/blob/7cb6a0d7c67585441c93eeb89103390e156f0319/.github/workflows/runtests.yml#L149-L151

**Comment by sumit-158** [comment]
Hey, @oscargus I have done the task please review it?

**Comment by oscargus** [comment]
Seems to work, but it would be better to have it on the same line. Now numpy is first installed, removed and reinstalled.

On the other hand, this limits the numpy-version, so not sure how it should be prioritized.

File: .github/workflows/runtests.yml [hunk 1]
```diff
@@ -149,6 +149,7 @@ jobs:
       - run: pip install mpmath gmpy2 matplotlib numpy scipy aesara ipython \
                          symengine cython llvmlite wurlitzer \
                          autowrap numexpr 'antlr4-python3-runtime==4.7.*'
+      - run: pip install numba
```
Comment (oscarbenjamin) [inline .github/workflows/runtests.yml:152]: This should just be added to the previous line

Comment (sumit-158) [reply .github/workflows/runtests.yml]: can you please review now!!

File: .github/workflows/runtests.yml [hunk 2]
```diff
@@ -148,8 +148,8 @@ jobs:
       - run: python -m pip install --upgrade pip
       - run: pip install mpmath gmpy2 matplotlib numpy scipy aesara ipython \
                          symengine cython llvmlite wurlitzer \
-                         autowrap numexpr 'antlr4-python3-runtime==4.7.*'
-      - run: pip install numba
+                         autowrap numexpr 'antlr4-python3-runtime==4.7.*' \ 
+                         pip install numba
```
Comment (oscargus) [inline .github/workflows/runtests.yml:152]:
```suggestion
                         numba
```

Comment (oscargus) [reply .github/workflows/runtests.yml:152]: These are package names. We do not want to install `pip` (as it is already installed) and not install `install` (as it is not needed).

File: .github/workflows/runtests.yml (hunk 1 above)
Comment (oscargus) [reply .github/workflows/runtests.yml]: Please refrain from using exclamation marks, especially multiple. If it took you ten days to fix it, why the hurry now?

**Comment by oscargus** [comment]
Can you please rebase and squash the commits? Right now there are nine commits for a rather marginal change. Thanks!

**Comment by oscargus** [comment]
Looks good! Thanks!

(We should probably also merge the more thorough tests from #21318 and #20088 in, with correct attribution.)

**Pull request merged**